ci/test/05_before_script.sh

```diff
@@ -33,7 +33,9 @@ if [ -z "$NO_DEPENDS" ]; then
   else
     SHELL_OPTS="CONFIG_SHELL="
   fi
-  DOCKER_EXEC $SHELL_OPTS make $MAKEJOBS -C depends HOST=$HOST $DEP_OPTS
+  # Temporary workaround for https://github.com/bitcoin/bitcoin/issues/16368
```

Comment:
It is a combination of all of the qt build issues I've been running into lately.

* Not sure how to pass in compiler flags to the qt build from outside to disable the wdeprecated warnings in depends. #15822
* Not sure if we can bump qt to a more recent version to get rid of the warnings #18580
* Not sure how to solve #16368

Reply:
I am just hoping that one of you build people will solve this one day :pray: kthxbye

To limit the changes here to purely the ci folder, I will leave any build changes for the future.

Reply:
To clarify, yes this commit is required. Otherwise the build log would exceed 40000 lines and be killed.

Reply:
> to the qt build from outside to disable the wdeprecated warnings in depends.

Can you add `NO_QT=1 NO_WALLET=1` to DEP_OPTS? Given qt is no-longer used here, and bdb wasn't used beforehand. If anything that should reduce the build output.

> Not sure if we can bump qt to a more recent version

That's blocked until c++17, unless someone wants to fix c++11 support for a newer version (5.12) of Qt for macOS.

Regardless of above, I'll try follow up here and improve this for you.

Reply:
> Can you add NO_QT=1 NO_WALLET=1 to DEP_OPTS? Given qt is no-longer used here, and bdb wasn't used beforehand. If anything that should reduce the build output.

Well the goal is to eventually enable tsan for the wallet and gui. I might wait for sqlite to avoid the bdb horror, but at least qt will be tested against hopefully soon.

Also, it is nice to have a test for the "can I compile qt depends with clang", even though the build result is thrown away.

Reply:
> That's blocked until c++17, unless someone wants to fix c++11 support for a newer version (5.12) of Qt for macOS.

Thanks, didn't know that they switched away from cxx11

Reply:
> Thanks, didn't know that they switched away from cxx11

heh. They didn't "officially". They just started using c++14 features in portions of the macOS code, while claiming to still support c++11. Some details in #17768.